src/test/kotlin/org/rust/ide/inspections/typecheck/ChangeReprAttributeFixTest.kt

```diff
@@ -0,0 +1,56 @@
+/*
+ * Use of this source code is governed by the MIT license that can be
+ * found in the LICENSE file.
+ */
+
+package org.rust.ide.inspections.typecheck
+
+import org.rust.ide.inspections.RsInspectionsTestBase
+import org.rust.ide.inspections.RsTypeCheckInspection
+
+class ChangeReprAttributeFixTest : RsInspectionsTestBase(RsTypeCheckInspection::class) {
+    fun `test change repr on enum`() = checkFixByText(
+        "Change representation of enum 'Foo' to '#[repr(u16)]'", """
+        #[repr(u8)]
+        enum Foo {
+            FooVariant = <error>1u16<caret></error>
+        }
+    """, """
+        #[repr(u16)]
+        enum Foo {
+            FooVariant = 1u16
+        }
+    """
+    )
+
+    fun `test add repr on enum`() = checkFixByText(
+        "Change representation of enum 'Foo' to '#[repr(u16)]'", """
+        enum Foo {
+            FooVariant = <error>1u16<caret></error>
+        }
+    """, """
+        #[repr(u16)]
+        enum Foo {
+            FooVariant = 1u16
+        }
+    """
+    )
+
+    fun `test don't offer fix for non-integral types`() = checkFixIsUnavailable(
+        "Change representation of enum 'Foo' to '#[repr(u16)]'", """
+        enum Foo {
+            FooVariant = <error>'a'<caret></error>
+        }
+    """
+    )
+
+    fun `test don't offer fix for aliased types`() = checkFixIsUnavailable(
```

Comment:
I'm not sure about this one. Maybe it's better to suggest changing the representation to `u8` in this case?

<img width="609" alt="Screen Shot 2021-12-01 at 06 04 33" src="https://user-images.githubusercontent.com/6079006/144164605-cc04b257-2215-4401-bb51-b1ab60daec99.png">

@Undin What do you think?